csvcubed/csvcubed/readers/v1_0/columnschema.py

```diff
@@ -0,0 +1,459 @@
+    ) -> NewQbDimension:
+
+        new_dimension = NewQbDimension.from_data(
+            label=self.label or label,
+            data=data,
+            description=self.description,
+            parent_dimension_uri=self.from_existing,
+            source_uri=self.definition_uri,
+        )
+        new_dimension.code_list = self._get_code_list(new_dimension, json_parent_dir)
+        return new_dimension
+
+    def _get_code_list(
+        self,
+        new_dimension: NewQbDimension,
+        json_parent_dir: Path,
+    ) -> Optional[QbCodeList]:
+
+        code_list_obj = None
+
+        if isinstance(self.code_list, str):
+            if looks_like_uri(self.code_list):
+                code_list_obj = ExistingQbCodeList(self.code_list)
+
+            else:
+                code_list_path = Path(self.code_list)
+                if code_list_path.is_absolute():
+                    code_list_obj = NewQbCodeListInCsvW(code_list_path)
+                else:
+                    code_list_obj = NewQbCodeListInCsvW(
+                        json_parent_dir / self.code_list
+                    )
+        elif isinstance(self.code_list, bool):
+            if self.code_list is False:
+                code_list_obj = None
+
+            elif (
+                new_dimension.parent_dimension_uri
+                == "http://purl.org/linked-data/sdmx/2009/dimension#refPeriod"
+                and self.definition_uri is not None
+                and self.definition_uri.lower().startswith("http://reference.data.gov.uk/id/")
+            ):
+                # This is a special case where we build up a code-list of the date/time values.
+                code_list_obj = self._get_date_time_code_list_for_dimension(
+                    self.label, new_dimension
+                )
+                # else, the user wants a standard codelist to be automatically generated
+                return code_list_obj or new_dimension.code_list
+        else:
+            raise ValueError(f"Unmatched code_list value {self.code_list}")
+
+        return code_list_obj or new_dimension.code_list
+
+    def _get_date_time_code_list_for_dimension(
+        self, column_title: str, new_dimension: NewQbDimension
+    ) -> CompositeQbCodeList:
+        csvw_safe_column_title = csvw_column_name_safe(column_title)
+        assert isinstance(new_dimension.code_list, NewQbCodeList)
+        return CompositeQbCodeList(
+            CatalogMetadata(new_dimension.label),
+            [
+                DuplicatedQbConcept(
+                    existing_concept_uri=uritemplate.expand(
+                        c.value,
+                        {csvw_safe_column_title: c.label},
+                    ),
+                    label=c.label,
+                    code=c.code,
+                )
+                for c in new_dimension.code_list.concepts
+            ],
+        )
+
+
+@dataclass
+class ExistingDimension(SchemaBaseClass):
+    from_existing: str = ""
```

Comment:
I don't think `from_existing` should have a default value if it's the only property which classifies it as being an `ExistingDimension`.